go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -1340,7 +1340,13 @@ func (nc *DefaultNodeNetworkController) reconcileConntrackUponEndpointSliceEvent
 		return fmt.Errorf("cannot reconcile conntrack: %v", err)
 	}
 	svc, err := nc.watchFactory.GetService(namespacedName.Namespace, namespacedName.Name)
-	if err != nil && !apierrors.IsNotFound(err) {
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.V(5).Infof("Service %s/%s not found (might have been deleted) when reconciling conntrack for endpointslice %s",
+				namespacedName.Namespace, namespacedName.Name, oldEndpointSlice.Name)
+			// TODO: service is not found, likely deleted, flushing service conntrack entries shall be handled at service reconciliation
+			return nil
+		}
 		return fmt.Errorf("error while retrieving service for endpointslice %s/%s when reconciling conntrack: %v",
 			newEndpointSlice.Namespace, newEndpointSlice.Name, err)
```

Comment:
This is actually my code, but since we're here already, could you fix the following? 
```suggestion
        return fmt.Errorf("error while retrieving service for endpointslice %s/%s when reconciling conntrack: %v",
            oldEndpointSlice.Namespace, oldEndpointSlice.Name, err)
```
Upon endpoint slice delete, we call `h.nc.reconcileConntrackUponEndpointSliceEvents(endpointslice, nil)` and if an error happens in `nc.watchFactory.GetService` above that is not `IsNotFound` , here we panic :-S